### DISCUSSION TITLE
ASGI callable returned without starting response.

### DISCUSSION THREAD

**Discussion by Serena-Xu** [fastapi/fastapi#6644] [Questions] [answered]
The fastAPI that I am working on does not return traceback whenever the request failed, instead, it returns 500 Internal Server Error with error :
```
ERROR:    ASGI callable returned without starting response.
2021-05-14 16:12:08 - uvicorn.error:409 - ERROR - ASGI callable returned without starting response.
```

Anyone experienced this problem before and know the fix to it ?

**Comment by allen-munsch** [comment]
- https://stackoverflow.com/questions/67540578/asgi-callable-returned-without-starting-response?noredirect=1#comment119380254_67540578
- https://gitter.im/tiangolo/fastapi?at=609edf42d523ee3d3e321d99

I'm not sure where exactly the issue should be raised, but it looks like the response is started but never disconnected?

- https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/http/httptools_impl.py#L412

```
elif not self.response_started and not self.disconnected:
```

**Comment by Serena-Xu** [comment, marked as answer]
I've found the issue caused this problem. It was due to a custom middleware in the API. I am closing this issue.

**Comment by tiangolo** [comment]
Thanks for the help here @allen-munsch ! 👏  🙇 

Thanks for reporting back and closing the issue @Serena-Xu  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
